gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/MultiEnvironmentsIT.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.gradle.plugin.tests;
+
+import net.minidev.json.parser.ParseException;
+import org.eclipse.jkube.kit.common.ResourceVerify;
+import org.gradle.testkit.runner.BuildResult;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@RunWith(Parameterized.class)
+public class MultiEnvironmentsIT {
+  @Rule
+  public final ITGradleRunner gradleRunner = new ITGradleRunner();
+
+  @Parameterized.Parameters(name = "environment {0}")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(
+        new Object[] { "common" },
+        new Object[] { "dev" },
+        new Object[] { "prod" },
+        new Object[] { "common,dev"},
+        new Object[] { "common,prod"}
+    );
+  }
+
+  @Parameterized.Parameter
+  public String environment;
+
+  @Test
+  public void k8sResource_whenRun_generatesK8sManifestsContainingConfigMap() throws IOException, ParseException {
+    // When
+    final BuildResult result = gradleRunner.withITProject("multi-environments")
+        .withArguments("build", "-Pjkube.environment=" + environment, "k8sResource", "--stacktrace")
+        .build();
+    // Then
+    ResourceVerify.verifyResourceDescriptors(gradleRunner.resolveDefaultKubernetesResourceFile(),
+        gradleRunner.resolveFile("expected", environment.replace(",", "and"), "kubernetes.yml"));
```

Comment:
This makes it hard to read. It's better if you define another parameter for the expected directory (just like we did for other tests `expectedDirectory`).

nit: Also, if you add the new parameter, it would be nice if the expected directories were renamed from `commonanddev` and `commonandprod` to `common-and-dev` and `common-and-prod` so they are more readable.